### ISSUE TITLE
Savon::QualifiedMessage translates nested :attributes! and :content! keys.

### ISSUE THREAD

**Issue by jmcnevin** [savonrb/savon#597]
It appears that QualifiedMessage does not treat these keys properly if you have a few levels of nesting happening, as I was seeing tags in my SOAP envelopes like tns:Attributes and tns:Content being created.

For now, I'm doing a quick monkey-patch like this, but being a relative newcomer to this gem, I haven't put together a real PR yet.

```
require 'savon'

# Fixes Savon to appropriately handle nested :attributes!/:content! usage.
module Savon
  class QualifiedMessage

    def to_hash(hash, path)
      return unless hash
      return hash.map { |value| to_hash(value, path) } if hash.kind_of?(Array)
      return hash.to_s unless hash.kind_of? Hash

      hash.inject({}) do |newhash, (key, value)|
        case key
        when :order!
          add_namespaces_to_values(value, path)
          newhash.merge(key => value)
        when :attributes!, :content!
          newhash.merge(key => to_hash(value, path))
        else
          translated_key = Gyoku.xml_tag(key, :key_converter => @key_converter).to_s
          translated_key << "!" if key[-1] == "!"

          newpath = path + [translated_key]

          if @used_namespaces[newpath]
            newhash.merge(
              "#{@used_namespaces[newpath]}:#{translated_key}" =>
                to_hash(value, @types[newpath] ? [@types[newpath]] : newpath)
            )
          else
            newhash.merge(translated_key => value)
          end
        end
      end
    end

  end
end
```

Anyone else experienced this?

**Comment by cadetstar** [comment]
+1, we're seeing it on something we're doing, too.

**Comment by misleb** [comment]
+1, also seeing this, 3 years later! This is a pretty serious bug. Without the monkey patch from @jmcnevin I'm unable to specify content! for tags that I need to also have \@attributes

**Comment by pcai** [comment]
This should be fixed in https://github.com/savonrb/savon/issues/597 which was released this week as v2.11.2. please reopen if you find issues.
